src/kixi/datastore/web_server.clj

```diff
@@ -262,37 +267,58 @@
    {:id :schema-create
     :methods
     {:post {:consumes "application/transit+json"
-            :response (fn [ctx]                  
-                        (let [name (get-in ctx [:parameters :path :name])
-                              body (get-in ctx [:body])]
-                          (c/submit communications
-                                    {::ss/name name
-                                     ::ss/definition (:definition body)})
-                          true))}
+            :produces "application/transit+json"
+            :response (fn [ctx]
```

Comment:
I've sketched out a multimethod alternative and it was a lot more code than this so tempted to leave for now. Dispatching on predicates isn't straightforward, unless I'm missing something.